core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
Thanks. It makes sense.

Just one more question: after a broker upgrades from 0.9 to 0.10, does broker need to update index file of the active segment? If so, should we keep the size of active segment's index file to be config.maxIndexSize?